### PR TITLE
INT-43 support toggling DNSSEC on zones

### PR DESCRIPTION
**Pull request by rupa**
* Ensure that leaving DNSSEC unset, or removing the setting, doesn't change
  existing value.
* pre-check for DNSSEC tests tries to create a zone with DNSSEC=true,
  if it fails due to account permissions, we skip the test.

### CODE REVIEW

**Comment by rupa** [comment]
NB: this depends on the changes to the Go SDK, so CI will likely fail until that's merge

File: ns1/resource_zone_test.go [hunk 1]
```diff
@@ -163,6 +172,90 @@ func TestAccZone_secondary(t *testing.T) {
...
+				),
+			},
+		},
+	})
+}
+
+// A Client instance we can use outside of a TestStep
+func sharedClient() (interface{}, error) {
+	var ignoreSSL bool
+	if os.Getenv("NS1_IGNORE_SSL") == "" {
```
[the remark this answers is not in the document]
Comment (rupa) [reply ns1/resource_zone_test.go]: I thiiiiiiiink this is right -  `Getenv` returns "" for either set-but-empty or unset, and in both cases we'd want `false`. Type in config is definitely `bool` so the shape of what it's set to doesn't matter (unless we have a bug in Config?)

File: ns1/resource_zone_test.go [hunk 2]
```diff
@@ -163,6 +172,90 @@ func TestAccZone_secondary(t *testing.T) {
...
+					testAccCheckZoneName(&zone, zoneName),
+					testAccCheckZoneDNSSEC(&zone, false),
+				),
+			},
+		},
+	})
+}
+
+// A Client instance we can use outside of a TestStep
+func sharedClient() (interface{}, error) {
```
[the remark this answers is not in the document]
Comment (rupa) [reply ns1/resource_zone_test.go]: woops yeah, some cargo-cult stuff from looking at how AWS does it

File: ns1/resource_zone_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mburtless) [reply ns1/resource_zone_test.go]: 💯 you're definitely right that we want `config.IgnoreSSL` set to `false` if `Getenv` returns the zero value.

What I was getting at is that if the env var was set like`export NS1_IGNORE_SSL=false`, then the config the provider generates would set `config.IgnoreSSL` to `false`.  As is, this code would set `config.IgnoreSSL` to `true`, right?

[the remark this answers is not in the document]
Comment (rupa) [reply ns1/resource_zone_test.go]: ah jeez yeah gotcha. As a shell nerd i kinda hate that someone would do that, but we should allow it. 

I'd like to put something in the docs to clarify, so we don't have to think about whether `NS1_IGNORE_SSL=purplemonkeydishwasher` should be `true` or `false`

**Pull request merged**